python/README.md

```diff
@@ -0,0 +1,187 @@
+## Getting started
+This module contains the building blocks to build new Flow connectors in Python, as well as adapting external Python connectors written in standard protocols to Flow. A Python connector is a subclass of the `Connector` classes in the `capture`, `derive`, and `materialize` (once it's written) submodules. Convenience classes `shim_airbyte_cdk.CaptureShim` and `shim_singer_sdk.CaptureShim` are provided to adapt existing Airbyte or Singer/Meltano connectors.
+
+A Python connector is composed of a few critical pieces:
+* `pyproject.toml` defines the Python module and its dependencies. For example:
+  ```toml
+  [tool.poetry.dependencies]
+  # If we're pulling in an open-source connector that doesn't need any changes, we can define it here. Subdirectory can be omitted if the connector lives at the root of the repo.
+  source_bigdata = { git = "https://github.com/big/data.git", subdirectory = "connector" }
+  # If we're working with an open-source connector that we have pulled in-tree, we can define it like this, where the connector was pulled into the subdirectory "upstream-connector"
+  source_bigdata = { path = "upstream-connector" }
```

Comment:
I think I'm 👎 on having the additional indirection of `upstream-connector`.

The extra level of nesting is awkward, and if we're making changes to it's implementation and behavior then it's no longer "upstream", in the sense that it's a unmodified & vendor'd chunk of code. And we pretty much _always_ intend to make _some_ modification to the upstream connector (for example, by simplifying its usually-incorrect schema and enabling schema inference).

Can we just have it be the current directory, as `source-pokemon` does ?

Reply:
How strongly do you feel about this? I ask because Python really wants its modules to be in subfolders. I tried changing to pulling the upstream connector directly into the `source-airtable` folder and that didn’t work because it’s a separate module that’s not configured like Poetry expects (i.e it has `setup.py` and requirements.txt instead of `pyproject.toml`). 

We _could_ tell people that when they import a connector, they also have to rejigger its python packaging to be consistent with all the rest, but given that it “just works” as a separate module imported by Poetry through `pyproject.toml`, I’m inclined to leave it be. Thoughts?

Reply:
That's fair. I hadn't considered the impacts of setup.py et al.

In that case, though, a more conventional layout would be `source-airtable/source_airtable`, right? Given that's the actual name of the module, and is also how it's organized in the upstream repo.

I was mostly reacting to the "upstream-connector" naming.